core/src/main/java/dk/alexandra/fresco/framework/sce/SecureComputationEngine.java

```diff
@@ -4,55 +4,78 @@
+   * </p>
    *
-   * @param application The application to evaluate.
-   * @param network the network to run the computations on
+   * @param application the application to evaluate
+   * @param resources the resource pool used in evaluation
+   * @param network the network
+   * @return output of the application
    */
-  <OutputT> OutputT runApplication(
-      Application<OutputT, Builder> application,
-      ResourcePoolT resources, Network network);
+  default <OutputT> OutputT runApplication(Application<OutputT, BuilderT> application,
+      ResourcePoolT resources, Network network) {
+    return runApplication(application, resources, network, DEFAULT_TIMEOUT);
+  }
+
+  /**
+   * Executes an application based on the current configuration.
+   *
+   * @param application the application to evaluate
+   * @param resources the resource pool used in evaluation
+   * @param network the network
+   * @param timeout time to wait for application output
+   * @return output of the application
+   */
+  <OutputT> OutputT runApplication(Application<OutputT, BuilderT> application,
+      ResourcePoolT resources, Network network, Duration timeout);
 
   /**
    * Executes an application based on the current SCEConfiguration. If the SecureComputationEngine
-   * is not setup before (e.g. connected to other parties etc.), the SecureComputationEngine will
-   * do the setup phase before running the application.
-   * <br/>
-   * In a normal application this should be the normal way to start an application since there
-   * need to be allocated resources (the resource pool) and allowed for parallel work.
+   * is not setup before (e.g. connected to other parties etc.), the SecureComputationEngine will do
+   * the setup phase before running the application. <br/>
+   * In a normal application this should be the normal way to start an application since there need
+   * to be allocated resources (the resource pool) and allowed for parallel work.
    *
    * @param application The application to evaluate.
+   * @param resources the resource pool used in evaluation
    * @param network the network to run the computations on
    * @return the future holding the result
    */
-  <OutputT> Future<OutputT> startApplication(
-      Application<OutputT, Builder> application,
+  <OutputT> Future<OutputT> startApplication(Application<OutputT, BuilderT> application,
       ResourcePoolT resources, Network network);
 
 
   /**
-   * Initializes the SecureComputationEngine. This method is idempotent - and sets up the
-   * proces queue and initializes the protocol suite.
+   * Initializes the SecureComputationEngine. This method is idempotent - and sets up the proces
+   * queue and initializes the protocol suite.
    */
   void setup();
 
   /**
-   * Ensures that resources are shut down properly. Network is disconnected
-   * and sockets are released.
+   * Ensures that resources are shut down properly. Network is disconnected and sockets are
+   * released.
```

Comment:
Network is no longer the responsibility of the SCE

Reply:
Ok, I will update the comment.